### DISCUSSION TITLE
[IntelliJ Integration] How to restore images (binary files) with `updateScene` or else ?

### DISCUSSION THREAD

**Discussion by bric3** [excalidraw/excalidraw#7553] [Q&A]
Hi,

I'm the developer of the [Excalidraw IntelliJ Platform Plugin](https://github.com/bric3/excalidraw-jetbrains-plugin), and I am not sure how one can restore images in the `files` field. I'm using the `updateScene`. Because having the files in the `initialData` is impractical as the client calls the API after Excalidraw has been opened.

I have noticed the `ExcallidrawAPI.addFiles`, however as I understand it this cannot work since ids will be different.

**Comment by innovatorved** [comment]
have you figure this out. i m experiencing the same problem

**Reply by bric3** [reply]
Nope, I'm not a frontend developer, I barely know TS or JS ; and reading the Excalidraw (0.17.0) code didn't got me very far to see if I have missed something.

**Comment by bric3** [comment]
I wonder if this comment could help 🤔

https://github.com/excalidraw/excalidraw/issues/7240#issuecomment-1812161000
